contracts/sol6/helpers/KyberRateQueryReserves.sol

```diff
@@ -0,0 +1,163 @@
+pragma solidity 0.6.6;
+
+import "@kyber.network/utils-sc/contracts/Utils.sol";
+import "@kyber.network/utils-sc/contracts/IERC20Ext.sol";
+import "../IConversionRates.sol";
+import "../IKyberSanity.sol";
+import "../IKyberReserve.sol";
+
+interface IKyberReserveExt is IKyberReserve {
+    function conversionRatesContract() external view returns (IConversionRates);
+    function sanityRatesContract() external view returns (IKyberSanity);
+}
+
+contract KyberRateQueryReserves is Utils {
+
+    function getRatesWithEth(
+        IKyberReserveExt reserve,
+        IERC20Ext[] calldata tokens,
+        uint256 weiAmount
+    )
+        external view returns(uint256[] memory sellRates, uint256[] memory buyRates)
+    {
+        uint256 numTokens = tokens.length;
+
+        buyRates = new uint256[](numTokens);
+        sellRates = new uint256[](numTokens);
+        for (uint256 i = 0; i < numTokens; i++) {
+            (buyRates[i], sellRates[i], , , ) = getRateWithEth(reserve, tokens[i], weiAmount);
+        }
+    }
+
+    function getRatesWithToken(
+        IKyberReserveExt reserve,
+        IERC20Ext[] calldata tokens,
+        uint256 tweiAmount
+    )
+        external view 
+        returns(uint256[] memory sellRates, uint256[] memory buyRates)
+    {
+        uint256 numTokens = tokens.length;
+
+        buyRates = new uint256[](numTokens);
+        sellRates = new uint256[](numTokens);
+        for (uint256 i = 0; i < numTokens; i++) {
+            (buyRates[i], sellRates[i], ,  , ) = getRateWithToken(reserve, tokens[i], tweiAmount);
+        }
+    }
+
+    function getReserveRates(
+        IKyberReserveExt reserve,
+        IERC20Ext[] calldata srcs,
+        IERC20Ext[] calldata dests
+    )
+        external view returns(uint256[] memory pricingRates, uint256[] memory sanityRates)
+    {
+        require(srcs.length == dests.length, "srcs length != dests");
+
+        pricingRates = new uint256[](srcs.length);
+        sanityRates = new uint256[](srcs.length);
+        IKyberSanity sanityRateContract;
+        IConversionRates conversionRateContract;
+
+        try reserve.sanityRatesContract() returns (IKyberSanity sanityContract) {
+            sanityRateContract = sanityContract;
+        } catch {
+            revert("no sanityRate contract");
```

Comment:
why do we add revert here? was thinking that we don't want to revert in case we want to fetch rates data for all reserves.

Reply:
Or is it only used for FPR/APR reserves?

Reply:
actually only used for fpr.
but I agree its better to be able to use it without sanity rate as well. so removed this revert.